packages/react/src/components/DatePicker/next/DatePicker.js

```diff
@@ -0,0 +1,648 @@
+        (
+          selectedDateElem ||
+          todayDateElem ||
+          calendarContainer.querySelector('.flatpickr-day[tabindex]') ||
+          calendarContainer
+        ).focus();
+      }
+    }
+
+    function handleOnChange() {
+      if (start.value !== '') {
+        return;
+      }
+
+      if (!calendar.selectedDates) {
+        return;
+      }
+
+      if (calendar.selectedDates.length === 0) {
+        return;
+      }
+
+      calendar.clear();
+      calendar.input.focus();
+    }
+
+    if (start) {
+      start.addEventListener('keydown', handleArrowDown);
+      start.addEventListener('change', handleOnChange);
+
+      // Flatpickr's calendar dialog is not rendered in a landmark causing an
+      // error with IBM Equal Access Accessibility Checker so we add an aria
+      // role to the container div.
+      calendar.calendarContainer.setAttribute('role', 'region');
+      // IBM EAAC requires an aria-label on a role='region'
+      calendar.calendarContainer.setAttribute(
+        'aria-label',
+        'calendar-container'
+      );
+    }
+
+    if (end) {
+      end.addEventListener('keydown', handleArrowDown);
+      end.addEventListener('change', handleOnChange);
+    }
+
+    //component did unmount equivalent
+    return () => {
+      // Note: if the `startInputField` ref is undefined then calendar will be
+      // of type: Array and `destroy` will not be defined
+      if (calendar && calendar.destroy) {
+        calendar.destroy();
+      }
+
+      if (start) {
+        start.removeEventListener('keydown', handleArrowDown);
+        start.removeEventListener('change', handleOnChange);
+      }
+
+      if (end) {
+        end.removeEventListener('keydown', handleArrowDown);
+        end.removeEventListener('change', handleOnChange);
+      }
+    };
+  }, [savedOnChange, savedOnClose, savedOnOpen]); //eslint-disable-line react-hooks/exhaustive-deps
+
+  // update calendar config if props change
+  // component did update equivalent
+  useEffect(() => {
+    if (calendarRef.current) {
+      calendarRef.current.set({ dateFormat });
+    }
+  }, [dateFormat]);
+
+  useEffect(() => {
+    if (calendarRef.current && minDate) {
+      calendarRef.current.set('minDate', minDate);
+    }
+  }, [minDate]);
+
+  useEffect(() => {
+    if (calendarRef.current && maxDate) {
+      calendarRef.current.set('maxDate', maxDate);
+    }
+  }, [maxDate]);
+
+  useEffect(() => {
+    if (calendarRef.current && disable) {
+      calendarRef.current.set('disbale', disable);
+    }
+  }, [disable]);
+
+  useEffect(() => {
+    if (calendarRef.current && enable) {
+      calendarRef.current.set('enable', enable);
+    }
+  }, [enable]);
+
+  useEffect(() => {
+    if (calendarRef.current && inline) {
+      calendarRef.current.set('inline', inline);
+    }
+  }, [inline]);
+
+  useEffect(() => {
+    if (calendarRef.current) {
+      calendarRef.current.set({ value });
+      updateClassNames(calendarRef.current);
```

Comment:
Is `updateClassNames` something that can be pulled out of the component or does it change based on data?

Reply:
@joshblack If I understand correctly, it changes based on data. It's used when the component mounts and also here.

Reply:
@jnm2377 to avoid the `react-hooks/exhaustive-deps` warnings (and prevent future bugs) we should pull out `updateClassNames` of the component and add any additional arguments that it needs / depends on in the scope.

This should remove the need to get rid of the `react-hooks/exhaustive-deps` warnings (at least in cases where it's because of `updateClassNames`

Reply:
@joshblack Is this a blocker for this PR? I looked into it yesterday afternoon, but was running into some issues bc it's used in the [useEffect](https://github.com/carbon-design-system/carbon/blob/1139c0e15820d3e023c93a764ef65fa777f094bb/packages/react/src/components/DatePicker/next/DatePicker.js#L269-L271) that creates the flatpickr calendar, and then it's used to actually create the flatpickr [calendar instance](https://github.com/carbon-design-system/carbon/blob/1139c0e15820d3e023c93a764ef65fa777f094bb/packages/react/src/components/DatePicker/next/DatePicker.js#L333-L340), so I'm not sure how that would work w/ pulling out the function. If it's working as expected, was wondering if this is ok as is?

Reply:
@jnm2377 the bug that we can run into is that if `updateClassNames` in the future is updated to use some kind of state then that state value would be stale (since the `useEffect` is only seeing the first "version" of this function).

What error were you seeing when you moved `updateClassNames` into the outermost scope of the page?

Reply:
@joshblack when I pull it out, it no longer creates the flatpickr calendar instance (from the initial useEffect), so then there are a bunch of additional errors from other parts of the code that reference the instance. I think I understand the potential issue you're describing. I guess, I feel like that's more of an enhancement to support a state.

Reply:
@jnm2377 I think what might be happening is that `updateClassNames` requires `prefix` (it's using it from the parent scope) and so when it's moved to the top-level scope things are erroring out, included the calendar creation that you're talking about.

If it is updated to include `prefix` as an argument I think you won't see that issue anymore, let me know!

---

For background on why this is something I noticed, in general, we do not want to disable the eslint rule for the rules of hooks. There are specific moments where that is okay (if we can prove the dependencies will always be up-to-date or missing deps are stable) but they should be isolated since ignoring the rules of hooks will lead to hard-to-find bugs later down the line (in my opinion)

Reply:
@joshblack I had included prefix differently (had also pulled it out to be used in the function), but hadn't tried including it as an argument. 👍🏽

Reply:
@joshblack pushed the changes, ready for review